Refactored Code into modules, fixed filters

Pushkalgithub opened the pull request:
closes #58
closes #49
closes #50 

Activated location filters
Only displayed latitude upto the number of recorded figures (Dont pad with 0)

Pushkalgithub committed:
Refactor filters into module and fix Downloads, Blank screens and Taxon View rendering

Pushkalgithub committed:
Fix all taxa filter not applying other filters + fix map rendering issue

Pushkalgithub committed:
5 modules created (App_info, citations, data table, taxon view and trait view), clear filter fixed

dfalster reviewed (changes requested):
Looks great overall, but clear filters button is not working.

Pushkalgithub commented:
Fixed the clear filters issue! Changes made:
- Removed duplicate clear filters logic that was causing the hang
- All filter inputs (trait name, basis of record, life stage, location) now clear properly
- Data table resets to show full dataset after clearing filters
- Added small delay to ensure smooth UI updates

Everythings hopefully working smoothly now

dfalster reviewed (approved):
Great work! Everything works nicely. And I really like the modular structure of the code. 

Feel free to merge. Please use the "Squash and Merge" option when merging into develop. That creates a single commit from everything on the branch
